src/main/java/httpserver/HttpServer.java

```diff
@@ -0,0 +1,48 @@
+package httpserver;
+
+import httpApplication.Application;
+import server.ConnectionDataStream;
+import server.Process;
+
+import java.io.IOException;
+import java.util.HashMap;
+
+public class HttpServer implements Process {
+    public HttpServer(Application application) {
+        this.application = application;
+    }
+
+    public void runWith(ConnectionDataStream client) {
+
+        try {
+            HttpIncomingMessage clientRequest = new HttpRequest(parseRawRequest(client.readInput()));
+            HttpResponse applicationResponse = application.start(clientRequest);
+
+            client.write(convertHttpResponseToRawRequest(applicationResponse));
+            client.close();
+
+        } catch (IOException ignore) {
+        }
+    }
+
+    private String convertHttpResponseToRawRequest(HttpResponse httpResponse) {
+        StringBuilder response = new StringBuilder();
+        String statusLine = "HTTP/1.1 " + httpResponse.status;
+
+        response.append(statusLine);
+
+        return response.toString();
+    }
```

Comment:
This feels like an extra responsibility. Something to keep an eye on for extraction later.